### PR TITLE
name SligSfx enum in AO and AE, name some elum funcs/fields and add Flags_2AA enum

### PR DESCRIPTION
**Pull request by mouzedrift**

### CODE REVIEW

File: Source/AliveLibAO/Abe.cpp [hunk 1]
```diff
@@ -1925,15 +1930,15 @@ void Abe::ElumFree_4228F0()
         }
     }
 
-    if ((field_2AA_flags >> 3) & 1)
+    if (field_2AA_flags.Get(Flags_2AA::e2AA_Bit2_bSfxPlaying))
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:1933]: this is 0x8

File: Source/AliveLibAO/Abe.cpp [hunk 2]
```diff
@@ -1925,15 +1930,15 @@ void Abe::ElumFree_4228F0()
...
     {
         ResourceManager::FreeResource_455550(ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, ResourceID::kElumUnknownResID_100, 1, 0));
         ResourceManager::FreeResource_455550(ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, ResourceID::kElumUnknownResID_112, 1, 0));
         ResourceManager::FreeResource_455550(ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, ResourceID::kElumUnknownResID_220, 1, 0));
         ResourceManager::FreeResource_455550(ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, ResourceID::kElumUnknownResID_221, 1, 0));
     }
 
-    if ((field_2AA_flags >> 4) & 1)
+    if (field_2AA_flags.Get(Flags_2AA::e2AA_Bit1))
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:1941]: this is 0x10

File: Source/AliveLibAO/Elum.cpp [hunk 3]
```diff
@@ -3718,7 +3717,14 @@ void Elum::VUpdate_4102A0()
             }
             
             (this->*sElum_motion_table_4C5148[field_FC_current_motion])();
-            
+
+            if (oldMotion != field_FC_current_motion &&
```
Comment (paulsapps) [inline Source/AliveLibAO/Elum.cpp:3721]: can prob delete this debug code if its no longer needed?

Comment (mouzedrift) [reply Source/AliveLibAO/Elum.cpp:3721]: still needed the 3 motions are still unknown

**Pull request merged**